services/microcredit/src/borrowers.ts

```diff
@@ -0,0 +1,55 @@
+import { database, utils, subgraph, config } from '@impactmarket/core';
+import { getAddress } from '@ethersproject/address';
+
+export async function updateBorrowers(): Promise<void> {
+    utils.Logger.info('Updating borrowers...');
+    const { subgraphMicroCreditBorrowers, appUser } = database.models;
+    const t = await database.sequelize.transaction();
+    try {
+        const lastUpdate = await subgraphMicroCreditBorrowers.findOne({
+            attributes: ['updatedAt'],
+            order: [['updatedAt', 'DESC']]
+        });
+
+        const entityLastUpdated = lastUpdate ? (lastUpdate.updatedAt.getTime() / 1000) | 0 : undefined;
```

Comment:
Ahh, I see what you are doing. But doing it this way will basically always fetch all users. The goal is to fetch only the ones updated in the last X minutes. One possible way is to save the timestamp of the last sync (update). It doesn't matter when each one of them was last updated. What matters is when was the last time that data was synced (fetched and updated successfully).